integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonInsertIntoCommand.scala

```diff
@@ -449,67 +443,48 @@ case class CarbonInsertIntoCommand(databaseNameOp: Option[String],
     }
   }
 
-  private def isAlteredSchema(tableSchema: TableSchema): Boolean = {
-    if (tableInfo.getFactTable.getSchemaEvolution != null) {
-      tableInfo
-        .getFactTable
-        .getSchemaEvolution
-        .getSchemaEvolutionEntryList.asScala.exists(entry =>
-        (entry.getAdded != null && entry.getAdded.size() > 0) ||
-        (entry.getRemoved != null && entry.getRemoved.size() > 0)
-      )
-    } else {
-      false
-    }
-  }
-
   def getReArrangedIndexAndSelectedSchema(
       tableInfo: TableInfo,
       partitionColumnSchema: mutable.Buffer[ColumnSchema]): (Seq[Int], Seq[ColumnSchema]) = {
     var reArrangedIndex: Seq[Int] = Seq()
     var selectedColumnSchema: Seq[ColumnSchema] = Seq()
-    var complexChildCount: Int = 0
     var partitionIndex: Seq[Int] = Seq()
-    val columnSchema = tableInfo.getFactTable.getListOfColumns.asScala
+    val internalOrderColumns: util.ArrayList[CarbonColumn] = new util.ArrayList[CarbonColumn]()
+    internalOrderColumns.addAll(table.getVisibleDimensions)
+    internalOrderColumns.addAll(table.getVisibleMeasures)
+    val columnSchema = internalOrderColumns.asScala.map(col => col.getColumnSchema)
```

Comment:
Can you do like following?
```
val columnSchema = (table.getVisibleDimensions.asScala ++ table.getVisibleDimensions.asScala).map(_.getColumnSchema)
```

Reply:
yes, can be done like this now. previously it was used for something else

Reply:
done